nodebuilder/module.go

```diff
@@ -26,11 +26,11 @@ func ConstructModule(tp node.Type, network p2p.Network, cfg *Config, store Store
 	log.Infow("Accessing keyring...")
 	ks, err := store.Keystore()
 	if err != nil {
-		fx.Error(err)
+		return fx.Error(err)
```

Comment:
I presume it reports the error to FX and then returns it, correct?

Reply:
provides an err to fx and fx checks that err before constructing app and if err != nil, return